rename.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import re
+import sys
+import tempfile
+import textwrap
+
+TEMPLATE_SNAKE = 'pulp_plugin_template'
+TEMPLATE_CAMEL = 'PulpPluginTemplate'
+TEMPLATE_CAMEL_SHORT = 'PluginTemplate'
+TEMPLATE_DASH = 'pulp-plugin-template'
+TEMPLATE_DASH_SHORT = 'plugin-template'
+IGNORE_DIRS = ['.git']
+IGNORE_FILES = ['.gitignore', 'LICENSE', 'rename.py', 'flake8.cfg']
+
+
+def is_valid(name):
+    """
+    Check if specified name is compliant with requirements for it
+    """
+    return bool(re.match(r'^[a-z][0-9a-z_]{2,15}$', name))
```

Comment:
Will do.
There are few reasons why I suggest to put at least some limit. One limit for us is a directory name - depends on filesystem but usually max filename length is 255. We also use plugin name in class names so for readability and to fit line length :) it makes sense to ask for something relatively short I think. And 16 characters seem reasonable to me and it can be easily changed by anyone in case they need more.

Reply:
Seems reasonable. I think it is helpful to add a comment though, or else future pulp engineers might assume there is a  correctness reason that it is limited to 16.